lib/postprocessing/image.js

```diff
@@ -192,7 +199,47 @@ async function read(infile) {
     return gm(`${infile}[0]`);
 }
 
-async function applyOperations(img, instructions) {
+function getImageSize(metadata) {
+    // The orientation corresponds to the TIFF orientation tag which ranges fromn 1 to 8
+    // Values 5 to 8 require a swap of width and height
+    // See https://www.awaresystems.be/imaging/tiff/tifftags/orientation.html or
+    // https://www.adobe.io/content/dam/udp/en/open/standards/tiff/TIFF6.pdf
+    if(CHANGE_ORIENTATION_LIST.includes(metadata.orientation)) {
+        return new Dimensions(metadata.pageGeometry.height, metadata.pageGeometry.width);
+    } else {
+        return new Dimensions(metadata.pageGeometry);
+    }
+}
+async function applyWatemark(img, rendition, directories) {
+    const instructions = rendition.instructions;
+    const targetBoundingBox = new Dimensions(instructions.width, instructions.height);
+
+    // TODO drop old watermark.widthPercent once no longer used by AEM
+    const watermarkScale = instructions.watermark.scale || (instructions.watermark.widthPercent / 100.0) || 1.0;
+
+    // determine image dimensions
+    const imageSize = getImageSize(rendition.intermediateMetadata);
+
+    // TODO drop old watermark.watermarkContent once no longer used by AEM
+    const watermarkSource = instructions.watermark.image || instructions.watermark.watermarkContent;
+    // download watermark
+    if(!watermarkSource) {
+        throw new GenericError(`post-processing: Missing watermark.image in instructions : ${JSON.stringify(rendition.instructions)}`);
```

Comment:
Nitpick: Do we need to log the full instructions?

Reply:
This is defensive and should not have happened as `api-process` should have failed this. we discussed this in mob and decided to keep the full rendition(in case it ever happens in tests)